transport/compress.go

```diff
@@ -28,12 +30,15 @@ type compr struct {
 	c lz4.Compressor
 }
 
-func (c *compr) compress(dst io.Writer, src *bytes.Buffer) (written int64, err error) {
+func (c *compr) compress(ctx context.Context, dst io.Writer, src *bytes.Buffer) (written int64, err error) {
 	c.buf = *src
 	buf := c.buf.Bytes()
 	read := len(c.buf.Bytes())
 
 	if read <= 9 {
+		if ctx.Err() != nil {
```

Comment:
wouldn't checking if context is done and returning wrapped error be better here?